WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java

```diff
@@ -445,168 +328,99 @@ private void renderPost(final int position, final ReaderPostViewHolder holder) {
-        } else {
-            holder.mTxtBlogUrl.setVisibility(View.GONE);
-            holder.mDotSeparator.setVisibility(View.GONE);
-        }
-
-        holder.mTxtDateline.setText(DateTimeUtils.javaDateToTimeSpan(post.getDisplayDate(), WordPress.getContext()));
-
-        if (post.hasBlogImageUrl()) {
-            String imageUrl = GravatarUtils.fixGravatarUrl(post.getBlogImageUrl(), mAvatarSzMedium);
-            mImageManager.loadIntoCircle(holder.mImgAvatarOrBlavatar, ImageType.BLAVATAR_CIRCULAR, imageUrl);
-            holder.mImgAvatarOrBlavatar.setVisibility(View.VISIBLE);
-        } else {
-            mImageManager.cancelRequestAndClearImageView(holder.mImgAvatarOrBlavatar);
-            holder.mImgAvatarOrBlavatar.setVisibility(View.GONE);
-        }
-
-        if (post.hasBlogName()) {
-            holder.mTxtAuthorAndBlogName.setText(post.getBlogName());
-        } else {
-            holder.mTxtAuthorAndBlogName.setText(null);
-        }
-
-        int imgFeaturedCornerRadius = holder.mImgFeatured
-            .getContext()
-            .getResources()
-            .getDimensionPixelSize(R.dimen.reader_featured_image_corner_radius);
-        if (post.getCardType() == ReaderCardType.PHOTO) {
-            // posts with a suitable featured image that have very little text get the "photo
-            // card" treatment - show the title overlaid on the featured image without any text
-            holder.mTxtText.setVisibility(View.GONE);
-            holder.mTxtTitle.setVisibility(View.GONE);
-            holder.mFramePhoto.setVisibility(View.VISIBLE);
-            holder.mTxtPhotoTitle.setVisibility(View.VISIBLE);
-            holder.mTxtPhotoTitle.setText(post.getTitle());
-            mImageManager.loadImageWithCorners(
-                holder.mImgFeatured,
-                ImageType.READER,
-                post.getFeaturedImageForDisplay(mPhotonWidth, mPhotonHeight),
-                imgFeaturedCornerRadius
-            );
-            holder.mThumbnailStrip.setVisibility(View.GONE);
-        } else {
-            mImageManager.cancelRequestAndClearImageView(holder.mImgFeatured);
-            holder.mTxtTitle.setVisibility(View.VISIBLE);
-            holder.mTxtTitle.setText(post.getTitle());
-            holder.mTxtPhotoTitle.setVisibility(View.GONE);
-
-            if (post.hasExcerpt()) {
-                holder.mTxtText.setVisibility(View.VISIBLE);
-                holder.mTxtText.setText(post.getExcerpt());
-            } else {
-                holder.mTxtText.setVisibility(View.GONE);
+        Function3<Long, Long, Boolean, Unit> onBookmarkClicked = (postId, blogId, aBoolean) -> {
+            toggleBookmark(post.blogId, post.postId);
+            notifyItemChanged(position);
+            return Unit.INSTANCE;
+        };
+        Function3<Long, Long, Boolean, Unit> onLikeClicked = (postId, blogId, aBoolean) -> {
+            toggleLike(ctx, post, position, holder);
+            return Unit.INSTANCE;
+        };
```

Comment:
Not sure what's the recommended practice to create click listener objects - is it fine to create new click listeners objects inside `onBindViewHolder`?  

I've seen them to be created inside VH init at places in the existing code for e.g. [here](https://github.com/wordpress-mobile/WordPress-Android/blob/2974e0111e0704d0feedece7a68472f74f429216/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainViewHolder.kt#L42).

Considering they're light objects, I think we might want to keep them here for better code readability. Lmkwyt.

Reply:
Great question! In general, it's better to create them in the init. However, I need access to the post, vh and context in these legacy onclick listeners. So I decided to keep them within the method - in case we notice any performance impact, which I doubt, we'll need to do a bit more refactoring.